cohttp-lwt-unix/src/io.ml

```diff
@@ -80,3 +80,33 @@ let catch f =
     | ex -> Lwt.fail ex)
 
 let pp_error = Fmt.exn
+
+let wait_eof_or_closed conn ic sleep_fn =
+  let open Lwt.Infix in
```

Comment:
The use of `Lwt.Infix` was for the `>|=` operator that is on line `97`.

Reply:
I think you should either

- (tradition) mimic the style of the library and use `>>=` followed by `return` instead of `>|=`, or
- (modernity) not mimic the style of the library and use `Lwt.Syntax`.

The current middle-ground doesn't seem particularly useful.